src/ci_providers/index.ts

```diff
@@ -0,0 +1,34 @@
+import * as providerAppveyorci from './provider_appveyorci'
```

Comment:
*This is in context to the entire file.*

This type of file pattern is called a *barrel*. One of the newer ways to structure a barrel is to have an `export * from […]` statement for each file from which you are re-exporting things.

Reply:
Note that barrel files cause a bit of a tangle for the TypeScript compiler to sift through so they shouldn't be used all that often. There's a good bit of value in using the pattern here because of how many objects are being exported in a uniform manner. Ideally, however, this would be the place to use a pattern called *inversion of control* to register each provider.

Reply:
I fully intend to change this, once I get things passing. I would like to create an interface that all providers implement for consistency .